src/NumericalAlgorithms/Spectral/ComplexDataView.cpp

```diff
@@ -0,0 +1,298 @@
+// reference.
+template <>
+ComplexDataView<ComplexRepresentation::Interleaved>::ComplexDataView(
+    const gsl::not_null<ComplexDataVector*> vector, const size_t size,
+    const size_t offset) noexcept
+    : size_{size},
+      real_slices_up_to_date_{false},
+      // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+      data_real_{reinterpret_cast<double*>(vector->data() + offset)},
+      // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+      data_imag_{data_real_ + 1} {
+  complex_view_.set_data_ref(vector->data() + offset, size);
+}
+
+// Due to the nature of this class performing manual memory management, indexing
+// into complex data with perscribed strides, it is necessary to violate type
+// system conventions to index into the complex data as though it is double
+// data, thus using a `reinterpret_cast`. The C++ standard guarantees that
+// complex numbers are represented in memory in such a way that this is correct,
+// despite the type subversion. We also cannot take the address of the
+// components of the complex data, as those are returned by value, not lvalue
+// reference.
+template <>
+ComplexDataView<ComplexRepresentation::Interleaved>::ComplexDataView(
+    std::complex<double>* const start, const size_t size) noexcept
+    : size_{size},
+      real_slices_up_to_date_{false},
+      // NOLINTNEXTLINE(cppcoreguidelines-pro-type-reinterpret-cast)
+      data_real_{reinterpret_cast<double*>(start)},
+      // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+      data_imag_{data_real_ + 1} {
+  complex_view_.set_data_ref(start, size);
+}
+
+// RealsThenImags constructors cause an allocation and a copy for each of the
+// real and imaginary components of the vector. If it becomes a performance
+// concern, the `real_slice_` and `imag_slice_` could be made as non-owning
+// DataVectors which reference a larger block of memory allocated only once.
+template <>
+ComplexDataView<ComplexRepresentation::RealsThenImags>::ComplexDataView(
+    const gsl::not_null<ComplexDataVector*> vector, const size_t size,
+    const size_t offset) noexcept
+    : size_{size},
+      real_slices_up_to_date_{true},
+      complex_view_{} {
+  complex_view_.set_data_ref(vector->data() + offset, size_);
+  real_slice_ = real(complex_view_);
+  imag_slice_ = imag(complex_view_);
+  data_real_ = real_slice_.data();
+  data_imag_ = imag_slice_.data();
+}
+
+template <>
+ComplexDataView<ComplexRepresentation::RealsThenImags>::ComplexDataView(
+    std::complex<double>* const start, const size_t size) noexcept
+    : size_{size},
+      real_slices_up_to_date_{true},
+      complex_view_{} {
+  complex_view_.set_data_ref(start, size_);
+  real_slice_ = real(complex_view_);
+  imag_slice_ = imag(complex_view_);
+  data_real_ = real_slice_.data();
+  data_imag_ = imag_slice_.data();
+}
+
+// For Interleaved views, assigning individual components is comparatively
```

Comment:
Would be clearer to replace "For Interleaved views," -> "For `ComplexDataView<Interleaved>`,"

Similarly elsewhere, where the phrase "Interleaved views" is used.

The phrase "For RealsThenImags views" is less confusing because "RealsThenImags" is not an English word.